controller/resource.go

```diff
@@ -99,54 +197,95 @@ func (t TaskLister) List() ([]Resource, error) {
 	return resources, nil
 }
 
-type Task struct {
+// fetchACLTokens retrieves all of the ACL tokens from Consul and
+// returns a mapping from service name to the ACL tokens that service has.
+func (s ServiceStateLister) fetchACLTokens() (map[string][]*api.ACLToken, error) {
+	aclTokens := make(map[string][]*api.ACLToken)
+
+	tokenList, _, err := s.ConsulClient.ACL().TokenList(nil)
+	if err != nil {
+		return aclTokens, err
+	}
+
+	for _, tokenEntry := range tokenList {
+		token, _, err := s.ConsulClient.ACL().TokenRead(tokenEntry.AccessorID, nil)
+
+		if err != nil {
+			return aclTokens, err
+		}
+
+		if len(token.ServiceIdentities) == 1 {
+			serviceName := token.ServiceIdentities[0].ServiceName
+			if _, ok := aclTokens[serviceName]; !ok {
+				aclTokens[serviceName] = []*api.ACLToken{token}
+			} else {
+				aclTokens[serviceName] = append(aclTokens[serviceName], token)
```

Comment:
This is an interesting case. In theory, we should never have multiple tokens for the same service since we always only create one per service. So if we found multiple tokens for the same service, they might be for a service from somewhere else (i.e. not this cluster).  

The question is what should we do in this case. The good thing is that I don't think we'll delete it because we expect that the secret in the AWS secrets manager exists for this token. 

Unfortunately, Consul does not allow you to add meta to the token, but maybe we can include cluster name in the token description and then parse it out to determine if the token is for this cluster.

I don't think this is blocking though since the original implementation also didn't handle this case properly.

Reply:
I added cluster name to the ACL token descriptions to address this